src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -0,0 +1,119 @@
+                => CommonCompletionItem.Create(
+                       displayText: SyntaxFacts.GetText(SyntaxKind.AwaitKeyword),
+                       displayTextSuffix: "",
+                       rules: CompletionItemRules.Default,
+                       Glyph.Keyword,
+                       description: RecommendedKeyword.CreateDisplayParts(SyntaxFacts.GetText(SyntaxKind.AwaitKeyword), string.Empty),
+                       inlineDescription: shouldMakeContainerAsync ? CSharpFeaturesResources.Make_container_async : null,
+                       isComplexTextEdit: shouldMakeContainerAsync);
+        }
+
+        internal override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem completionItem, TextSpan completionListSpan, char? commitKey, bool disallowAddingImports, CancellationToken cancellationToken)
+        {
+            // IsComplexTextEdit is true when we want to add async to the container.
+            if (!completionItem.IsComplexTextEdit)
+            {
+                return await base.GetChangeAsync(document, completionItem, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+            }
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var declaration = root.FindToken(completionListSpan.Start).GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            if (declaration is null)
+            {
+                return await base.GetChangeAsync(document, completionItem, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+            }
+
+            var documentWithAsyncModifier = document.WithSyntaxRoot(root.ReplaceNode(declaration, AddAsyncModifier(document, declaration)));
+            var formattedDocument = await Formatter.FormatAsync(documentWithAsyncModifier, Formatter.Annotation, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            using var _ = ArrayBuilder<TextChange>.GetInstance(out var builder);
+
+            builder.AddRange(await formattedDocument.GetTextChangesAsync(document, cancellationToken).ConfigureAwait(false));
+            builder.Add(new TextChange(completionListSpan, completionItem.DisplayText));
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var newText = text.WithChanges(builder);
+            return CompletionChange.Create(CodeAnalysis.Completion.Utilities.Collapse(newText, builder.ToImmutableArray()));
+
+            static SyntaxNode AddAsyncModifier(Document document, SyntaxNode declaration)
+            {
+                var generator = SyntaxGenerator.GetGenerator(document);
+                var modifiers = generator.GetModifiers(declaration);
+                return generator.WithModifiers(declaration, modifiers.WithAsync(true)).WithAdditionalAnnotations(Formatter.Annotation);
```

Comment:
this is a bit of a pity, needing to format everything.  the modifier will have elastic trivia on it.  so why not just find the elastic region and format that instead?

Reply:
@CyrusNajmabadi I have tried avoiding formatting here when I initially implemented this, but things didn't work as expected (or I couldn't get it working).

Reply:
so... this is the C# layer.  i think we shoudl jsut make the token, find the place for it, add it simply, without formatting.  the thing here is that we are literally just adding `async `.  So i think it might be better to make that literally as raw an edit as possible with nothign else.

Reply:
@CyrusNajmabadi I have tried with:

```csharp
var asyncToken = SyntaxFactory.Token(SyntaxKind.AsyncKeyword).WithTrailingTrivia(SyntaxFactory.Space);
                return declaration switch
                {
                    MethodDeclarationSyntax method => method.WithModifiers(method.Modifiers.Add(asyncToken)),
                    LocalFunctionStatementSyntax local => local.WithModifiers(local.Modifiers.Add(asyncToken)),
```

But things doesn't go well for cases like:

```csharp
public void M()
{
    void Local() => awa$$
}
```

It gets turned into something like `async     void Local()`. So I'll have to copy trivia around, and at the end, I believe some cases where things aren't handled correctly will occur.